src/pages/edge/adobe-journey-optimizer/code-based/tabs/api-reference.md

```diff
@@ -15,7 +15,31 @@ public static void getPropositionsForSurfaces(@NonNull final List<Surface> surfa
 * _surfaces_ is a list of surfaces for which propositions are requested.
 * _callback_ `call` method is invoked with propositions map of type `Map<Surface, List<Proposition>>`. If the callback is an instance of [AdobeCallbackWithError](../../../../home/base/mobile-core/api-reference.md#adobecallbackwitherror), and if the operation times out or an error occurs in retrieving propositions, the `fail` method is invoked with the appropriate [AdobeError](../../../../home/base/mobile-core/api-reference.md#adobeerror).
 
-**Syntax**
+**Example**
+
+#### Kotlin
+
+```kotlin
+val surface1 = Surface("myActivity#button")
+val surface2 = Surface("myActivityAttributes")
+val surfaces = listOf(surface1, surface2)
+
+Messaging.getPropositionsForSurfaces(surfaces) {
+  it?.let { propositionsMap ->
+           if (propositionsMap.isNotEmpty()) {
+             // get the propositions for the given surfaces
+             propositionsMap[surface1]?.let {
+               // read surface1 propositions
+             }
+             propositionsMap[surface2]?.let {
+               // read surface2 propositions
+             }
+           }
+          }
+}
```

Comment:
the formatting looks off - is it github?

Reply:
yeah... i can try to manually make it look better though